inst/include/plant/patch.h

```diff
@@ -89,13 +88,31 @@ class Patch {
   void r_compute_environment() {compute_environment();}
   void r_compute_rates() {compute_rates();}
 
+  // Prototype env. cache for assembly
+  std::vector<double> step_history{0.0};  // always start at zero
+  std::vector<std::vector<environment_type>> environment_history;
+
+  void cache_ode_step();
+  void cache_RK45_step(int step);
+  void load_ode_step();
+  std::vector<environment_type> environment_cache;
+
+  // use cache for mutant runs
+  bool save_RK45_cache;
+  bool use_cached_environment = false;
+
+  bool is_mutant_run = false;
+
+  void set_mutant();
+  void add_strategies(std::vector<strategy_type> strategies);
+  void overwrite_strategies(std::vector<strategy_type> strategies);
+
 private:
   void compute_environment();
   void rescale_environment();
   void compute_rates();
 
   parameters_type parameters;
-  std::vector<bool> is_resident;
   environment_type environment;
```

Comment:
if you follow my suggestions below about making environment a pointer, that would start here.

Reply:
Actually, might be environment_type *environment;